odoo/models.py

```diff
@@ -568,6 +577,11 @@ class BaseModel(metaclass=MetaModel):
     """
     __slots__ = ['env', '_ids', '_prefetch_ids']
 
+    env: api.Environment
+    id: int | NewId | typing.Literal[False]
```

Comment:
Maybe we should type this one as `IdType`.

Reply:
`IdType | typing.Literal[False]`